src/earn/saga.ts

```diff
@@ -1,14 +1,177 @@
 import { PayloadAction } from '@reduxjs/toolkit'
-import { depositStart, depositSuccess } from 'src/earn/slice'
+import BigNumber from 'bignumber.js'
+import erc20 from 'src/abis/IERC20'
+import { EarnEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { depositCancel, depositError, depositStart, depositSuccess } from 'src/earn/slice'
 import { DepositInfo } from 'src/earn/types'
 import { navigateHome } from 'src/navigator/NavigationService'
+import { CANCELLED_PIN_INPUT } from 'src/pincode/authentication'
+import { vibrateError } from 'src/styles/hapticFeedback'
+import { getTokenInfo } from 'src/tokens/saga'
+import { BaseStandbyTransaction } from 'src/transactions/actions'
+import { TokenTransactionTypeV2, newTransactionContext } from 'src/transactions/types'
+import Logger from 'src/utils/Logger'
+import { ensureError } from 'src/utils/ensureError'
 import { safely } from 'src/utils/safely'
-import { put, takeLeading } from 'typed-redux-saga'
+import { publicClient } from 'src/viem'
+import { getPreparedTransactions } from 'src/viem/preparedTransactionSerialization'
+import { sendPreparedTransactions } from 'src/viem/saga'
+import networkConfig, { networkIdToNetwork } from 'src/web3/networkConfig'
+import { all, call, put, takeLeading } from 'typed-redux-saga'
+import { decodeFunctionData } from 'viem'
+
+const TAG = 'earn/saga'
 
 export function* depositSubmitSaga(action: PayloadAction<DepositInfo>) {
-  // todo(act-1178): submit the tx
-  yield* put(depositSuccess())
-  navigateHome()
+  const { tokenId, preparedTransactions: serializablePreparedTransactions, amount } = action.payload
+
+  const preparedTransactions = getPreparedTransactions(serializablePreparedTransactions)
+
+  const tokenInfo = yield* call(getTokenInfo, tokenId)
+  if (!tokenInfo) {
+    Logger.error(`${TAG}/depositSubmitSaga`, 'Token info not found for token id', tokenId)
+    yield* put(depositError())
+    return
+  }
+
+  const networkId = tokenInfo.networkId
+  const trackedProperties = {
+    tokenId,
+    networkId,
+    tokenAmount: amount,
+    providerId: 'aave-v3',
+  }
+
+  let submitted = false
+
+  try {
+    Logger.debug(
+      `${TAG}/depositSubmitSaga`,
+      `Starting deposit for token ${tokenId}, total transactions: ${preparedTransactions.length}`
+    )
+
+    const createDepositStandbyTxHandlers = []
+
+    if (preparedTransactions.length > 1 && preparedTransactions[0].data) {
+      const { functionName, args } = yield* call(decodeFunctionData, {
```

Comment:
probably dumb q: what's the reason for not calling `decodeFunctionData` directly?

Reply:
just an optimization I guess? we only need this if there's more than one tx. Just copied from swaps